### ISSUE TITLE
WidgetTypeAround should use insertContent in order to integrate with track changes

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5#7229]
## 📝 Provide a description of the improvement

It uses `writer.insert()` while it should use `model.insertContent()`.

In addition, it should probably copy the implementation of the current \<kbd>Enter\</kbd> handler that we have for widgets. AFAICS it does some more checks.

---

If you'd like to see this improvement implemented, add a 👍 reaction to this post.

**Comment by Reinmar** [comment]
PS. I'm not entirely sure whether it's possible to insert a new empty paragraph with `insertContent()`. It may get merged automatically to the previous block. Let me know if that happens.

**Comment by Reinmar** [comment]
PPS. The Enter/Shift+Enter support uses `writer.insert()`, however it works with track changes because there's a custom integration for these two handlers. If we'll use `writer.insertContent()` for WidgetTypeAround (**if it can be used**), then track changes will integrate well automatically.

**Comment by Reinmar** [comment]
I asked @ckeditor/qa-team to help us verify that it will resolve the issue on the CF side. Here, let's focus on whether we can use insertContent() as I'm not sure about that unfortunately.

**Comment by Reinmar** [comment]
I talked with @Mgsy and it turned out that Enter isn't handled by track changes (so, there's no custom handler apparently) and that using insertContent() does not solve this issue.

Let's wait for @scofalik.

**Comment by scofalik** [comment]
There is integration for enter command and there is integration for `insertContent()`. See `integrations/entercommand.js` and `trackchangesediting.js`.

However, there's some kind of an error if you simply add a single paragraph :thinking:

```javascript
editor.model.change( writer => {
	const p = writer.createElement( 'paragraph' );
	editor.model.insertContent( p, writer.createPositionAt( editor.model.document.getRoot(), 0 ) );
} );
```

Most probably, the range is set in an incorrect way. I can't guarantee that we will be able to fix that in this iteration.

**Comment by Mgsy** [comment]
> There is integration for enter command and there is integration for `insertContent()`. See `integrations/entercommand.js` and `trackchangesediting.js`.

Note to clarify: Enter command is handled properly, however, not in a context of widgets - after pressing Enter/Shift+Enter, a new paragraph is inserted, instead of a suggestion.

**Comment by scofalik** [comment]
I see @oleq that you created `InsertParagraphCommand`. I assume you plan to use it also for enter/shift+enter behaviour when selection is on widget?

This can work. When you are done, could you please add an issue in collaboration features repository so we can work on it and have it done hopefully this iteration?

**Comment by oleq** [comment]
In a F2F talk with @scofalik we decided:

*   to go with the implementation as in #7237,
*   because the Track Changes integration with the `insertParagraph` command will take some time and it is not clear whether it is possible in this iteration, I will create a quick fix that hides the type around UI in TC mode:
    *   `WidgetTypeAround` plugin will listen to `InsertParagraphCommand#isEnabled` (which is `false` in TC mode) and add a CSS class to the root that will make the type around buttons disappear
    *   we will revert the CSS class approach in the next iteration when a proper integration arrives

**Comment by oleq** [comment]
FYI: The quick fix landed in 005a5d2.
